lib/modules/dosomething/dosomething_campaign_run/dosomething_campaign_run.module

```diff
@@ -296,3 +297,32 @@ function dosomething_campaign_run_form_campaign_run_node_form_alter(&$form, &$fo
 
   drupal_add_css(drupal_get_path('module', 'dosomething_campaign_run') . '/campaign_run_node.css');
 }
+
+/*
+ * When campaign run nodes are saved, this hook updates the campaign status field on campaign nodes
+ * based on the campaign run dates.
+ *
+ * Implements hook_node_presave().
+ */
+function dosomething_campaign_run_node_presave($node) {
+  if ($node->type == 'campaign_run') {
+    $run_campaigns = $node->field_campaigns;
+    $run_dates = $node->field_run_date;
+
+    // Determine the active and closed state for each translation of the run_date field
+    foreach ($run_dates as $lang => $date) {
+      $start = new DateTime($date[0]['value']);
+      $end = new DateTime($date[0]['value2']);
+      $status = dosomething_helpers_get_campaign_status($start, $end);
+      $run_campaigns = $node->field_campaigns[$lang];
+
+      // Update the campaign status field on each campaign node tied to this run.
+      // There could be multiple campaigns per campaign run. Probably not, but we do allow for it.
+      foreach ($run_campaigns as $campaign) {
```

Comment:
since this little chunk is the same here and in the cron, it could be a helper function `_dosomething_campaign_set_campaign_status`